modules/spark-wallet.nix

```diff
@@ -5,18 +5,17 @@ with lib;
 let
   cfg = config.services.spark-wallet;
   inherit (config) nix-bitcoin-services;
-  onion-chef-service = (if cfg.onion-service then [ "onion-chef.service" ] else []);
 
   # Use wasabi rate provider because the default (bitstamp) doesn't accept
   # connections through Tor
   torRateProvider = "--rate-provider wasabi --proxy socks5h://${config.services.tor.client.socksListenAddress}";
   startScript = ''
-    ${optionalString cfg.onion-service ''
-      publicURL="--public-url http://$(cat /var/lib/onion-chef/spark-wallet/spark-wallet)"
+    ${optionalString (cfg.getAnnounceAddressCmd != "") ''
+      publicURL="--public-url http://$(${cfg.getAnnounceAddressCmd})"
```

Comment:
`getAnnounceAddressCmd` doesn't seem to be a very fitting in this context especially. Why not just `getOnionAddressCmd`?

Reply:
See https://github.com/fort-nix/nix-bitcoin/pull/293#issuecomment-758131410.